dbms/src/Common/MemoryTracker.h

```diff
@@ -112,7 +112,9 @@ class MemoryTracker : public std::enable_shared_from_this<MemoryTracker>
     /// next should be changed only once: from nullptr to some value.
     void setNext(MemoryTracker * elem)
     {
-        next.store(elem, std::memory_order_relaxed);
+        MemoryTracker * old_val = nullptr;
+        if (!next.compare_exchange_strong(old_val, elem, std::memory_order_seq_cst, std::memory_order_relaxed))
```

Comment:
in what cases that `next` will be called multiple times？

Reply:
For example:
https://github.com/pingcap/tiflash/blob/5609e68c0fa213e9740ff52ec60eac97c884e45a/dbms/src/Interpreters/ProcessList.cpp#L168

Different query may belong to the same user, the the user's memory tracker will be call multiple times.

Reply:
Yes, total_memory_tracker is the same. But even if total_memory_tracker is the same, there is a write condition race when concurrent write to `next_holder`, that is why I made this change.